### PR TITLE
Add a utility class to check if we're on mobile or not, and then when we...

### PR DESCRIPTION
**Pull request by melissanoelle**
On mobile, when this is not present, the keyboard pops up all over the place, which pushes the content up half the window. It's a pretty normal UX to expect to tap in an input box before typing, and switching to a chatroom where you want to read some context first but having to first clear your keyboard and then scroll can be very annoying.

### CODE REVIEW

File: src/util.js [hunk 1]
```diff
@@ -358,6 +358,36 @@ Candy.Util = (function(self, $){
 		return ie;
 	};
 
+	/** Class: isMobile
+		* Checks to see if we're on a mobile device.
```
Comment (mweibel) [inline src/util.js]: Your indenting is slightly off here

**Comment by mweibel** [comment]
Hmm, we might need to check for something else instead: document width. This ensures that we don't miss any user agent or do the user agent wrong. 
Also, if we do user agent checks, it might be better to include a small library for that, no?

**Comment by melissanoelle** [comment]
Document width isn't reliable because of people who prefer to minimize their screen to really thin, which I think is a legitimate use-case.  

I would argue against including a small library when we have plain JS that will work just as well for us. `navigator.userAgent` is very well supported: https://developer.mozilla.org/en-US/docs/Web/API/NavigatorID.userAgent so we can trust that the `userAgent` won't be wrong.

File: src/view/pane/room.js [hunk 2]
```diff
@@ -336,7 +336,9 @@ Candy.View.Pane = (function(self, $) {
       if (pane) {
         // IE8 will fail maybe, because the field isn't there yet.
         try {
-          pane.children('.field')[0].focus();
+          if (!Candy.Util.isMobile.any()) {
```
Comment (benlangfeld) [inline src/view/pane/room.js]: Could you move this conditional to a short-circuit at the start of the method?

```javascript
setFocusToForm: function(roomJid) {
  if (Candy.Util.isMobile.any()) { return true; }
  ...
```

**Comment by benlangfeld** [comment]
@melissanoelle To be more comprehensive, and use what I believe to be the most up-to-date list and source, could you please use the snippet from http://detectmobilebrowsers.com/?

File: src/util.js [hunk 3]
```diff
@@ -358,6 +358,36 @@ Candy.Util = (function(self, $){
 		return ie;
 	};
 
+	/** Class: isMobile
+	  * Checks to see if we're on a mobile device.
+	  */
+
```
Comment (benlangfeld) [inline src/util.js]: Nuke this line to ensure the docs link up?

File: src/view/pane/room.js [hunk 4]
```diff
@@ -332,6 +332,9 @@ Candy.View.Pane = (function(self, $) {
      *   (String) roomJid - Room JID
      */
     setFocusToForm: function(roomJid) {
+      // If we're on mobile, don't focus the input field.
+      if (Candy.Util.isMobile.any()) { return true; }
```
Comment (benlangfeld) [inline src/view/pane/room.js]: Make this false since we're refusing to comply.

**Pull request merged**